docs/develop/java/message-passing.mdx

```diff
@@ -593,6 +593,12 @@ The SDK will then ensure that your constructor receives the Workflow input argum
 The Workflow input arguments are also passed to your `@WorkflowMethod` method -- that always happens, whether or not you use the `@WorkflowInit` annotation.
 This is useful if you have message handlers that need access to Workflow input: see [Initializing the Workflow first](/handling-messages#workflow-initializers).
 
+:::caution
+
+Do not make blocking calls from within your `@WorkflowInit` method. This could result in your Workflow being incompletely initialized at the start, meaning, for example, that Signal and Update handler registration would be delayed.
```

Comment:
```suggestion
Do not make blocking calls from within your `@WorkflowInit` method. This could result in your Workflow being incompletely initialized at the start, meaning, for example, that Signal, Query, and Update handler registration would be delayed.
```